src/components/interactive-builder/add-question-modal.component.tsx

```diff
@@ -580,6 +604,76 @@ const AddQuestionModal: React.FC<AddQuestionModalProps> = ({
                   />
                 </RadioButtonGroup>
               ) : null}
+
+              {questionType === 'programState' && (
+                <>
+                  <div>
+                    <TextInput
+                      id="program"
+                      labelText="Add program uuid"
+                      value={programUuid}
+                      onChange={(event: React.ChangeEvent<HTMLInputElement>) => {
+                        setProgramUuid(event.target.value), mutateProgram;
+                      }}
+                      required
+                    />
+                    <p className={styles.bodyShort01}>
+                      {programLookupError?.message && t('programUuidNotFound', 'Program uuid not found')}
+                    </p>
+                    <p className={styles.bodyShort01}>{program?.uuid ? program?.name : ''}</p>
+
+                    <TextInput
+                      id="programWorkflow"
+                      labelText="Add program workflow uuid"
+                      value={programWorkflowUuid}
+                      onChange={(event: React.ChangeEvent<HTMLInputElement>) => {
+                        setProgramWorkflowUuid(event.target.value), mutateProgramWorkflow;
+                      }}
+                      required
+                    />
+                    <p className={styles.bodyShort01}>
+                      {programWorkflowLookupError?.message &&
+                        t('programWorkflowUuidNotFound', 'Program workflow uuid not found')}
+                    </p>
+                    <p className={styles.bodyShort01}>
+                      {programWorkflow?.uuid ? programWorkflow?.concept.display : ''}
+                    </p>
+                  </div>
+                  <div>
+                    <FormLabel className={styles.label}>
+                      {t('searchForBackingProgramState', 'Search for a backing program state')}
+                    </FormLabel>
+                    {isLoadingProgramWorkflow && (
+                      <InlineLoading
+                        className={styles.loader}
+                        description={t('searchingProgramStates', 'Searching program states') + '...'}
+                      />
+                    )}
+                    {programWorkflowLookupError ? (
```

Comment:
Can we also add this logic to the Edit question modal so it's possible to change these values later on via the interactive builder?